MQTTClient-C/src/MQTTClient.h

```diff
@@ -73,35 +89,121 @@ extern void TimerCountdownMS(Timer*, unsigned int);
 extern void TimerCountdown(Timer*, unsigned int);
 extern int TimerLeftMS(Timer*);
 
+/**
+ * @brief Data structure for holding information about a message.
+ * 
+ */
 typedef struct MQTTMessage
 {
-    enum QoS qos;
+#if defined(MQTTV5)
+    /// @brief The MQTTv5 message properties.
+    MQTTProperties* properties;
+#endif /* MQTTV5 */
+    /// @brief The MQTT message QoS.
+    enum MQTTQoS qos;
+    /// @brief The MQTT message retained flag.
     unsigned char retained;
+    /// @brief The MQTT message dup flag.
     unsigned char dup;
+    /// @brief The MQTT message identifier.
     unsigned short id;
+    /// @brief The MQTT message payload.
     void *payload;
+    /// @brief The MQTT message `payload` length.
     size_t payloadlen;
 } MQTTMessage;
 
+/**
+ * @brief Data structure for holding information about a received message.
+ * 
+ */
 typedef struct MessageData
 {
+    /// @brief The MQTT message.
     MQTTMessage* message;
+    /// @brief The topic to which the message was published.
     MQTTString* topicName;
 } MessageData;
 
+/**
+ * @brief Data structure containing CONNACK information.
+ * 
+ */
 typedef struct MQTTConnackData
 {
+#if defined(MQTTV5)
+    /// @brief The MQTTv5 message properties.
+    MQTTProperties* properties;
+    /// @brief The MQTTv5 reason code.
+    enum MQTTReasonCodes reasonCode;
```

Comment:
@icraggs and others, I'm not certain about replacing the existing `unsigned char rc` with the enum: while it makes it easier to discover and understand the returned codes, I believe `sizeof(enum) > sizeof(char)`.

Reply:
@CIPop there should certainly be two variables, I think. ReasonCode for MQTT5. While sizeof(enum) > sizeof(char) we know that in this case the number of enums is not going to exceed the capacity of the one byte, so the conversion should work fine? If it's preferred to not use an enum because of the conversion, we could just use a list of constants instead (while retaining the enum info as comments perhaps).

Reply:
Sounds good: `reasonCode` as an `enum` makes the available values a lot more discoverable.

I wasn't aware of this but some compilers can optimize the size (even if it is non-C standard): e.g. [-fshort-enums](https://developer.arm.com/documentation/dui0774/e/Compiler-Command-line-Options/-fshort-enums---fno-short-enums)